cryptoaead/crypto_aead_aes256gcm_test.go

```diff
@@ -0,0 +1,70 @@
+package cryptoaead
+
+import (
+	"testing"
+	"bytes"
+	"github.com/google/gofuzz"
+)
+
+var testCount = 1000000
+
+type Test struct {
+	Message    []byte
+	Ad         []byte
+	Key        [32]byte
+	Nonce      [12]byte
+}
+
+func TestCryptoAEADAES256GCM(t *testing.T) {
+	// Skip the test if unsupported on this platform
+	if !CryptoAEADAES256GCMIsAvailable() {
+		t.Skip()
```

Comment:
This should log an warning for unsupported platforms. In case people assume it worked? Thoughts?